core/src/main/java/edu/wpi/grip/core/operations/composite/ContoursReport.java

```diff
@@ -44,4 +56,77 @@ public int getRows() {
     public int getCols() {
         return this.cols;
     }
+
+    /**
+     * This is called when the list of contours changes to indicate that the bounding boxes must be computed again
+     */
+    private void invalidateBoundingBoxes() {
+        boundingBoxes = Optional.empty();
+    }
+
+    /**
+     * Compute the bounding boxes of all contours (if they haven't already been computed).  Bounding boxes are used
+     * to compute several different properties, so it's probably not a good idea to compute them over and over again.
+     */
+    private List<Rect> computeBoundingBoxes() {
+        if (!boundingBoxes.isPresent()) {
+            List<Rect> bb = new ArrayList<>();
```

Comment:
Since this is being reallocated every time and this could be called upwards of 30 times per second this should be allocated with the size of `contours.size`